### PR TITLE
graph util

### PR DESCRIPTION
**Pull request by dominikstrak2001**
[wip]

### CODE REVIEW

File: readwriter.h [hunk 1]
```diff
@@ -374,11 +404,20 @@ class Graph {
     }    
 };
 
-class Tree : Graph {
-    
-
-
+void setupDirectories() {
+    for (const auto& dir : dirs) {
+        std::filesystem::create_directory(dir.second);
```
Comment (JakubPiotrowicz) [inline readwriter.h]: Status check

File: readwriter.h [hunk 2]
```diff
@@ -3,6 +3,17 @@
...
 #include "testlib.h"
 #include "bits/stdc++.h"
+#include <filesystem>
+
+enum PrintFormat {
+    Prompt,
+    Solution
+};
+
+const std::map<std::string, std::string> dirs = {
```
Comment (JakubPiotrowicz) [inline readwriter.h:17]: What is this map for?

**Review by JakubPiotrowicz — COMMENTED** [review]
Add comments

File: readwriter.h (hunk 2 above)
Comment (dominikstrak2001) [reply readwriter.h:17]: Added more descriptive values. Added Comment

File: readwriter.h (hunk 1 above)
Comment (dominikstrak2001) [reply readwriter.h]: Added status checking and comment

**Review by JakubPiotrowicz — APPROVED** [review]

**Pull request merged**